backend/app/controllers/spree/admin/stock_items_controller.rb

```diff
@@ -43,21 +43,24 @@ def load_stock_management_data
         @stock_locations = Spree::StockLocation.accessible_by(current_ability)
         @stock_item_stock_locations = params[:stock_location_id].present? ? @stock_locations.where(id: params[:stock_location_id]) : @stock_locations
         @variant_display_attributes = self.class.variant_display_attributes
-        @variants = Spree::Config.variant_search_class.new(params[:variant_search_term], scope: variant_scope).results
-        @variants = @variants.includes(:images, stock_items: :stock_location, product: :variant_images)
-        @variants = @variants.includes(option_values: :option_type)
-        @variants = @variants.order(id: :desc).page(params[:page]).per(params[:per_page] || Spree::Config[:orders_per_page])
+        @variants = Spree::Config.variant_search_class.new(params[:variant_search_term], scope: variant_scope).results.
+            order(id: :desc).page(params[:page]).per(params[:per_page] || Spree::Config[:orders_per_page])
       end
 
       def variant_scope
-        scope = Spree::Variant.accessible_by(current_ability)
-        if @product
-          scope = scope.where(
-            product: @product,
-            is_master: !@product.has_variants?
+        scope = Spree::Variant
+          .accessible_by(current_ability)
+          .distinct.order(:sku)
```

Comment:
@softr8 👋 We are having some issues with this `distinct` call. Getting `for SELECT DISTINCT, ORDER BY expressions must appear in select list (PG::InvalidColumnReference)`. Do you remember the reason why this was added?

Reply:
@kennyadsl @softr8 getting the same issue here while searching by variant name. 

```
ActiveRecord::StatementInvalid (PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: ...e_option_values"."name" ILIKE '%blish%') ORDER BY "spree_var...
```

will open an issue as this is a regression we introduced in 4.2